compiler/rustc_target/src/callconv/aarch64.rs

```diff
@@ -27,18 +32,40 @@ where
 
         let valid_unit = match unit.kind {
             RegKind::Integer => false,
-            RegKind::Float => true,
+            RegKind::Float => cx.target_spec().abi != "softfloat",
             RegKind::Vector => size.bits() == 64 || size.bits() == 128,
         };
 
         valid_unit.then_some(Uniform::consecutive(unit, size))
     })
 }
 
+fn softfloat_float_abi<Ty>(target: &Target, arg: &mut ArgAbi<'_, Ty>) {
+    if target.abi != "softfloat" {
+        return;
+    }
+    // Do *not* use the floag registers for passing arguments, as that would make
```

Comment:
nit: I think this is supposed to be `float`

```suggestion
    // Do *not* use the float registers for passing arguments, as that would make
```